packages/pwa-buildpack/lib/cli/create-project.js

```diff
@@ -0,0 +1,215 @@
+
+async function makeDirFromNpmPackage(packageName) {
+    const nameCheck = isValidNpmName(packageName);
+    if (typeof nameCheck === 'string') {
+        throw new Error(
+            `Invalid template: interpreted "${packageName}" as NPM package, which was invalid for the following reasons:\n${nameCheck}`
+        );
+    }
+    const packageDir = resolve(tmpDir, packageName);
+    // NPM extracts a tarball to './package'
+    const packageRoot = resolve(packageDir, 'package');
+    try {
+        if ((await fse.readdir(packageRoot)).includes('package.json')) {
+            prettyLogger.info(`Found ${packageName} template in cache`);
+            return packageRoot;
+        }
+    } catch (e) {
+        // Not cached.
+    }
+    let tarballUrl;
+    try {
+        prettyLogger.info(`Finding ${packageName} tarball on NPM`);
+        tarballUrl = JSON.parse(
+            execa.shellSync(`npm view --json ${packageName}`, {
+                encoding: 'utf-8'
+            }).stdout
+        ).dist.tarball;
+    } catch (e) {
+        throw new Error(
+            `Invalid template: could not get tarball url from npm: ${e.message}`
+        );
+    }
+
+    let tarballStream;
+    try {
+        prettyLogger.info(`Downloading and unpacking ${tarballUrl}`);
+        tarballStream = (await fetch(tarballUrl)).body;
+    } catch (e) {
+        throw new Error(
+            `Invalid template: could not download tarball from NPM: ${
+                e.message
+            }`
+        );
+    }
+
+    await fse.ensureDir(packageDir);
+    return new Promise((res, rej) => {
+        const untarStream = tar.extract({
+            cwd: packageDir
+        });
+        tarballStream.pipe(untarStream);
+        untarStream.on('finish', () => {
+            prettyLogger.info(`Unpacked ${packageName}`);
+            res(packageRoot);
+        });
+        untarStream.on('error', rej);
+        tarballStream.on('error', rej);
+    });
+}
+
+async function findTemplateDir(templateName) {
+    const template = templateAliases[templateName] || {
+        npm: templateName,
+        dir: templateName
+    };
+    try {
+        await fse.readdir(template.dir);
+        prettyLogger.info(`Found ${templateName} directory`);
+        // if that succeeded, then...
+        return template.dir;
+    } catch (e) {
+        prettyLogger.info(`No template at ${template.dir}.`);
+        return makeDirFromNpmPackage(template.npm);
+    }
+}
+
+module.exports.sampleBackends = require('../../sampleBackends.json');
+
+module.exports.command = 'create-project <directory>';
+
+module.exports.describe =
+    'Create a PWA project in <directory> based on template.';
+
+module.exports.builder = yargs =>
+    yargs
+        .version()
+        .showHelpOnFail(false)
+        .positional('directory', {
+            describe:
+                'Name or path to a directory to create and fill with the project files. This directory will be the project root.',
+            normalize: true
```

Comment:
This command was new to me, but it tells `yargs` to run `path.normalize` on the input which, if empty, will default to `.`

Just an FYI for others reading this code :)